estimatey-api/Estimatey.Infrastructure/DevOps/WorkItemSynchronizationService.cs

```diff
@@ -27,21 +28,31 @@ internal class WorkItemSynchronizationService
 
     public async Task SyncProject(ProjectEntity project)
     {
-        _logger.LogInformation("Syncing work items for project {projectName}.", project.DevOpsProjectName);
+        var semaphore = _semaphores.GetOrAdd(project.Id, _ => new SemaphoreSlim(1, 1));
+        await semaphore.WaitAsync();
+
+        try
+        {
+            _logger.LogInformation("Syncing work items for project {projectName}.", project.DevOpsProjectName);
 
-        var workItemRevisionsBatch = await _devOpsClient.GetWorkItemRevisionsBatch(project.DevOpsProjectName, project.WorkItemsContinuationToken);
+            var workItemRevisionsBatch = await _devOpsClient.GetWorkItemRevisionsBatch(project.DevOpsProjectName, project.WorkItemsContinuationToken);
 
-        await ProcessWorkItemRevisions(project, workItemRevisionsBatch);
+            await ProcessWorkItemRevisions(project, workItemRevisionsBatch);
 
-        var recycleBinWorkItems = await _devOpsClient.GetRecycleBinWorkItems(project.DevOpsProjectName);
+            var recycleBinWorkItems = await _devOpsClient.GetRecycleBinWorkItems(project.DevOpsProjectName);
 
-        await ProcessRecycleBinWorkItems(recycleBinWorkItems);
+            await ProcessRecycleBinWorkItems(recycleBinWorkItems);
 
-        _logger.LogInformation("Successfully synced work items for project {projectName}.", project.DevOpsProjectName);
+            _logger.LogInformation("Successfully synced work items for project {projectName}.", project.DevOpsProjectName);
 
-        if (!workItemRevisionsBatch.IsLastBatch)
+            if (!workItemRevisionsBatch.IsLastBatch)
+            {
+                await SyncProject(project);
+            }
+        }
+        finally
         {
-            await SyncProject(project);
+            semaphore.Release();
```

Comment:
And removing from `ConcurrentDictionary` here too

Reply:
Same as other, no big deal I don't think so leaving it.